### PR TITLE
update TypeScript results with 'downlevelIteration' option

### PR DESCRIPTION
**Pull request by trotyl**
Following https://github.com/kangax/compat-table/issues/919#issuecomment-396485252

Env:

+ TypeScript 3.0.1 (last rebase);
+ IE 9 in VirtualBox;

Note: 

+ Removed note `typescript.asyncawait` as generator is already supported under this flag;
+ Some `eval()` tests in es2016+ async function are left to be `null` (not sure how to test it without compilation);
+ There's a special case (`spreading non-iterables is a runtime error`) works before (accidentally) but not with this flag (not sure how to mark it);

Off-topic question:

Some subclassing tests are also passed (not related to this flag), is there any other requirements for transpilers? (Seems all of them are marked unsupported in current table)

image

### CODE REVIEW

File: data-common.json [hunk 1]
```diff
@@ -27,10 +27,15 @@
...
-    "asyncawait": {
+    "downlevelIteration": {
       "val": true,
-      "note_id": "typescript-async-await",
-      "note_html": "TypeScript <code>async</code> / <code>await</code> requires native generators support."
+      "note_id": "typescript-downlevel-iteration",
+      "note_html": "Requires the <code>downlevelIteration</code> compile option."
+    },
+    "notDownlevelIteration": {
+      "val": true,
```
Comment (zloirock) [inline data-common.json]: I don't care should this value be `true` with a note or `false` with a note. Let's listen to the opinions of the rest maintainers.

Comment (ljharb) [reply data-common.json]: I would expect false; trues in compilers are for default unflagged things.

Comment (trotyl) [reply data-common.json]: Done.

**Review by chicoxyzzy — APPROVED** [review]

**Review by ljharb — APPROVED** [review]

**Comment by ljharb** [comment]
This needs a rebase and a regen of the test data.

**Comment by trotyl** [comment]
Rebase done. (Re-checked on TypeScript 3.0.1, no data change)

**Comment by chicoxyzzy** [comment]
Thank you!

**Pull request merged**